action/context.go

```diff
@@ -0,0 +1,59 @@
+package action
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/caarlos0/env"
+)
+
+type Context struct {
+	Repository string `json:"repo" env:"GITHUB_REPOSITORY"`
+	SHA        string `json:"sha" env:"GITHUB_SHA"`
+	Ref        string `json:"ref" env:"GITHUB_REF"`
+	EventName  string `json:"eventName" env:"GITHUB_EVENT_NAME"`
+	Workflow   string `json:"workflow" env:"GITHUB_WORKFLOW"`
+	Action     string `json:"action" env:"GITHUB_ACTION"`
+	Actor      string `json:"actor" env:"GITHUB_ACTOR"`
+	Job        string `json:"job" env:"GITHUB_JOB"`
+	RunNumber  int    `json:"runNumber" env:"GITHUB_RUN_NUMBER"`
+	RunID      int    `json:"runID" env:"GITHUB_RUN_ID"`
+	ApiURL     string `json:"apiUrl" env:"GITHUB_API_URL" envDefault:"ttps://api.github.com"`
+	ServerURL  string `json:"serverUrl" env:"GITHUB_SERVER_URL" envDefault:"https://github.com"`
+	GraphQLUrl string `json:"graphqlUrl" env:"GITHUB_GRAPHQL_URL " envDefault:"https://api.github.com/graphql"`
+	Event      *Event `json:"payload"`
+	EventPath  string `env:"GITHUB_EVENT_PATH"`
+}
+
+func NewContextFromEnv() (*Context, error) {
+	ghContext := &Context{}
+	if err := env.Parse(ghContext); err != nil {
+		return nil, fmt.Errorf("parsing environments variables: %w", err)
+	}
+	if err := payloadFromFile(ghContext); err != nil {
+		return nil, fmt.Errorf("loading event from file: %w", err)
+	}
+	//TODO: handle INPUTS
```

Comment:
Should this be done before the code is merged?

Reply:
Removed the inputs comment